sigstore/_verify.py

```diff
@@ -236,27 +246,91 @@ def verify(
 
         logger.debug("Successfully verified signature...")
 
-        # Retrieve the relevant Rekor entry to verify the inclusion proof and SET
-        entry = self._rekor.log.entries.retrieve.post(
-            signature.decode(),
-            sha256_artifact_hash,
-            base64.b64encode(certificate).decode(),
-        )
-        if entry is None:
-            return RekorEntryMissing(
-                signature=signature.decode(), sha256_artifact_hash=sha256_artifact_hash
+        entry: Optional[RekorEntry]
+        if offline_rekor_entry is not None:
+            # TODO(ww): This should all go in a separate API, probably under the
+            # RekorEntry class.
+            logger.debug("offline Rekor entry: checking consistency")
+
+            try:
+                entry_body = json.loads(base64.b64decode(offline_rekor_entry.body))
+            except Exception:
+                return VerificationFailure(
+                    reason="couldn't parse offline Rekor entry's body"
+                )
+
+            # The Rekor entry's body should be a hashedrekord object.
+            # TODO: This should use a real data model, ideally generated from
+            # Rekor's official JSON schema.
+            kind, version = entry_body.get("kind"), entry_body.get("apiVersion")
+            if kind != "hashedrekord" or version != "0.0.1":
```

Comment:
out of scope for this, is there a tracking issue for supporting other types?

Reply:
https://github.com/sigstore/sigstore-python/issues/252